view_user_birthdate.py

```diff
@@ -0,0 +1,242 @@
+        assert valid_month(make_int(month)), "Invalid month."
+        assert valid_day(make_int(day), make_int(month)), "Invalid day."
+        assert valid_year(make_int(year)), "Invalid year."
+        self.month = month
+        self.day = day
+        self.year = year
+
+class AmericanDate(Date):
+    """Create an object of type AmericanDate wich inherits from class
+    Date.
+    """
+    nationalities = ['american', 'belizean', 'micronesian']
+    def format(self):
+        """Using the attributes 'month', 'day', and 'year', return a
+        string in the American format:
+        mm/dd/yyyy
+        """
+        return '%s/%s/%s' % (self.month, self.day, self.year)
+
+class EuropeanDate(Date):
+    """Create an object of type EuropeanDate which inherits from class
+    Date.
+    """
+    nationalities = ['british', 'irish', 'german', 'italian',
+                     'spanish', 'australian', 'new zealander',
+                     'south african', 'brazilian', 'mexican',
+                     'indian', 'indonesian', 'nigerian',
+                     'russian', 'canadian', 'french'] # Abbreviated list.
+    def format(self):
+        """Using the attributes 'month', 'day', and 'year', return a
+        string in the European format:
+        dd/mm/yyyy
+        """
+        return '%s/%s/%s' % (self.day, self.month, self.year)
+
+class ChineseDate(Date):
+    """Create an object of type ChineseDate wich inherits from class
+    Date.
+    """
+    nationalities = ['chinese', 'north korean', 'south korean',
+                     'japanese', 'taiwanese', 'hungarian',
+                     'lithuanian', 'iranian'] # Abbreviated list.
+    def format(self):
+        """Using the attributes 'month', 'day', and 'year', return a
+        string in the Chinese format:
+        yyyy/mm/dd
+        """
+        return '%s/%s/%s' % (self.year, self.month, self.day)
+
+def read_file(file_name):
+    """Read in data from a file and return it."""
+    with open(file_name, 'r') as some_file:
+        data = some_file.read()
+    return data
+
+def chunk_data(file_data, size):
+    """Take in data as a string and an integer as a string. Return a
+    list containing chunks of data that are 'size' number of lines
+    each.
+    """
+    data_lines = file_data.splitlines()
+
+    assert data_lines, "There is no data!"
+    assert int(size), "An integer in the form of a string must be provided."
+    assert 0 < int(size) <= len(data_lines), "A valid size must be provided."
+    assert len(data_lines) % size == 0, "Data is incomplete."
+
+    return [data_lines[pos:pos + size] for pos in range(0, len(data_lines), size)]
+
+def parse(data_chunk):
+    """Take in a chunk of data 3 lines long. Parse the 3 lines of code
+    into their corresponding parts: username, nationality, and
+    birthdate. Then return the parsed parts.
+    """
+    # Parse username.
+    first_line = re.search(r"username:(?P<username>[\w.-]+)", data_chunk[0])
+    if first_line:
```

Comment:
this repeats a common pattern - can you refactor this into a common function, something like the signature

def parse_chunk(regex, regex_group, chunk)

ex:

parse_chunk(r"username: ...", 'username', 0)